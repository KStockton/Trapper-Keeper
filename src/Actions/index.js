export const allNotes = notes => ({
  type: "ALL_NOTES",
  notes
});

export const deleteNote = id => ({
  type: "DELETE_NOTE",
  id
})
