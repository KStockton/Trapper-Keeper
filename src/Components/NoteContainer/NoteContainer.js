import React, { Component } from "react";
// import "./NoteContainer.scss";
import NoteCard from '../NoteCard/NoteCard';

class NoteContainer extends Component {
  render() {
    return (
      <div className="note-container-component">
        <NoteCard />
      </div>
    );
  }
}

export default NoteContainer;
