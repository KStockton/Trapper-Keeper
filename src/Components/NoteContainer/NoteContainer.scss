.note-container-component {
  width: 1200px;
  margin: 20px auto;
  columns: 4;
  column-gap: 40px;
}

@media (max-width: 1200px)
{
  .note-container-component {
    columns: 3;
    width: calc(100% - 40px);
    box-sizing: border-box;
    padding: 20px 20px 20px 0;
  }
}

@media (max-width: 768px)
{
  .note-container-component {
    columns: 2;
  }
}

@media (max-width: 480px)
{
  .note-container-component {
    columns: 1;
  }
}