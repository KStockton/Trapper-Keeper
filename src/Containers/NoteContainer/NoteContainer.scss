@import "../../Sass/Variables.scss";

.note-container-component {
  column-gap: 40px;
  columns: 4;
  margin: 20px auto;
  width: 1200px;
}

.note-container-start {
  background-color: #ffff88;
  box-shadow: 3px 10px 10px -10px rgba(0, 0, 0, 0.75);
  height: 19rem;
  margin: 150px auto;
  width: calc(100% - 57rem);
  min-width: 17rem;
  
  .intro-title {
    padding-top: 30px;
  }

  .intro-paragraph {
    margin: 0 auto;
    width: 80%;
  }
}

@media (max-width: 1200px)
{
  .note-container-component {
    columns: 3;
    width: calc(100% - 40px);
    box-sizing: border-box;
    padding: 20px 20px 20px 0;
  }
}

@media (max-width: 768px)
{
  .note-container-component {
    columns: 2;
  }
}

@media (max-width: 480px)
{
  .note-container-component {
    columns: 1;
  }
}