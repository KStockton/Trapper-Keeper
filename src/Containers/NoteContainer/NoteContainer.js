import React, { Component } from "react";
import NoteCard from "../NoteCard/NoteCard";
import { fetchAllNotes } from "../../Api/fetch/fetchAllNotes";
import { connect } from "react-redux";
import { allNotes } from "../../Actions/index";
import PropTypes from "prop-types";

export class NoteContainer extends Component {
  componentDidMount() {
    fetchAllNotes().then(results => this.props.allNotes(results));
  }

  renderNotes = () => {
    return this.props.notes
      .flat()
      .map(note => <NoteCard key={note._id} data={note} />);
  };

  render() {
    const { notes } = this.props;
    return (
      <React.Fragment>
        {notes.length ? (
          <div className="note-container-component">{this.renderNotes()}</div>
        ) : (
          <div className="note-container-start">
            <h3 className="intro-title">Welcome to TrapperKeeper</h3>
            <p className="intro-paragraph">
              TrapperKeeper is an application that helps you keep track of your
              busy life by creating online checklists. <br />
              <br /> Add a card by clicking the button in the upper right to
              start.
            </p>
          </div>
        )}
      </React.Fragment>
    );
  }
}

export const mapStateToProps = state => ({
  notes: state.notes
});

export const mapDispatchToProps = dispatch => ({
  allNotes: notes => dispatch(allNotes(notes))
});

NoteContainer.propTypes = {
  notes: PropTypes.array,
  allNotes: PropTypes.func
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NoteContainer);
