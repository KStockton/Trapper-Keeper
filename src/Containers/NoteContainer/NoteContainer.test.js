import React from "react";
import { shallow } from "enzyme";
import { NoteContainer, mapStateToProps, mapDispatchToProps } from './NoteContainer';
import { allNotes } from "../../Actions/index";


describe("NoteContainer", () => {
  let wrapper;
  let mockFn;
  let mockNotes;
  beforeEach(() => {
    mockNotes = [{
      id: 1, title: "testing", tasks:[{id: 77, text:"testing"}]
    },{
      id: 3, title: "testing2", tasks:[{id: 55, text:"testing2"}]
    }]
    mockFn = jest.fn()
    wrapper = shallow(<NoteContainer renderNotes={mockFn} notes ={mockNotes} />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should match the snapshot", () => {
    let notes = []
    wrapper = shallow(<NoteContainer notes={notes}/>)
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should set props as an object with an array from state', () => {
      const mockState = {
        notes: [{id:1, title: "Testing mapState", tasks:[{id:99, text:"testing"}]}],
        fakeState: "Fake for testing"
      }
      const expected = {
        notes: [{id:1, title: "Testing mapState", tasks:[{id:99, text:"testing"}]}],
      }

      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  });

  describe('mapDispatchToProps', () => {
 
    
    it('should call dispatch on allNotes', () => {
      const mockDispatch = jest.fn()
      const mockState = {
        notes: [{id:1, title: "Testing mapState", tasks:[{id:99, text:"testing"}]}],
        fakeState: "Not real state to return"
      }
      const actionToDispatch = allNotes(mockState)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.allNotes(mockState)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  });
});
