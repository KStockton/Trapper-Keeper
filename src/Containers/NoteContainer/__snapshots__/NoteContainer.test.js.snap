// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteContainer should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`NoteContainer should match the snapshot 2`] = `ShallowWrapper {}`;
